src/stream/src/executor/managed_state/aggregation/string_agg.rs

```diff
@@ -144,12 +151,12 @@ impl<S: StateStore> ManagedStringAggState<S> {
     ) -> StreamExecutorResult<()> {
         debug_assert!(super::verify_batch(ops, visibility, columns));
 
-        for (i, op) in ops.iter().enumerate() {
-            let visible = visibility.map(|x| x.is_set(i).unwrap()).unwrap_or(true);
-            if !visible {
-                continue;
-            }
-
+        for (i, op) in ops
+            .iter()
+            .enumerate()
+            .filter(|(i, _)| visibility.map(|x| x.is_set(*i).unwrap()).unwrap_or(true))
+            .filter(|(i, _)| columns[self.upstream_agg_col_idx].datum_at(*i).is_some())
```

Comment:
May add some comments here to make it clearer. :)

```suggestion
            .filter(|(i, _)| visibility.map(|x| x.is_set(*i).unwrap()).unwrap_or(true)) // skip invisible
            .filter(|(i, _)| columns[self.upstream_agg_col_idx].datum_at(*i).is_some()) // skip null input
```

Reply:
okk